### PR TITLE
Fix links to safekeeper protocol docs.

### PR DESCRIPTION
**Pull request by hlinnaka**
safekeeper/README_PROTO.md was moved to docs/safekeeper-protocol.md in
commit 0b14fdb078, as part of reorganizing the docs into 'mdbook' format.

Fixes issue #1475. Thanks to @banks for spotting the outdated references.

In addition to fixing the above issue, this patch also fixes other broken links as a result of 0b14fdb078. See https://github.com/neondatabase/neon/pull/2188#pullrequestreview-1055918480.

### CODE REVIEW

**Review by aome510 — APPROVED** [review]
LGTM. It appears that we have some other broken links:

https://github.com/neondatabase/neon/blob/3652cb7fb4d3941fb620cc61c4d5956b22125895/docs/sourcetree.md?plain=1#L31

https://github.com/neondatabase/neon/blob/3652cb7fb4d3941fb620cc61c4d5956b22125895/docs/sourcetree.md?plain=1#L60

https://github.com/neondatabase/neon/blob/3652cb7fb4d3941fb620cc61c4d5956b22125895/docs/SUMMARY.md?plain=1#L55

https://github.com/neondatabase/neon/blob/3652cb7fb4d3941fb620cc61c4d5956b22125895/docs/SUMMARY.md?plain=1#L58

https://github.com/neondatabase/neon/blob/fix-links-to-safekeeper-protocol/docs/pageserver-services.md?plain=1#L71

Not link but outdated references:

https://github.com/neondatabase/neon/blob/fix-links-to-safekeeper-protocol/docs/pageserver-services.md?plain=1#L121

https://github.com/neondatabase/neon/blob/3652cb7fb4d3941fb620cc61c4d5956b22125895/docs/glossary.md?plain=1#L78

**Comment by aome510** [comment]
> It appears that we have some other broken links:

Maybe we can squeeze these changes in this PR

**Comment by hlinnaka** [comment]
> > It appears that we have some other broken links:
> 
> Maybe we can squeeze these changes in this PR

Thanks, could you add a commit here, to fix those too, please?

**Comment by aome510** [comment]
> > > It appears that we have some other broken links:
> > 
> > 
> > Maybe we can squeeze these changes in this PR
> 
> Thanks, could you add a commit here, to fix those too, please?

Updated. Can you take look? I'm not really familiar with the renaming. I only made the changes based on https://github.com/neondatabase/neon/commit/0b14fdb0783b745fbb99e08d49da1253db2b0134.

> https://github.com/neondatabase/neon/blob/fix-links-to-safekeeper-protocol/docs/pageserver-services.md?plain=1#L71

I don't see any references nor renaming of ./src/remote_storage/README.md. Do u have any ideas?

File: docs/SUMMARY.md [hunk 1]
```diff
@@ -52,10 +52,10 @@
 - [multitenancy.md](./multitenancy.md) — how multitenancy is organized in the pageserver and Zenith CLI.
 - [settings.md](./settings.md)
 #FIXME: move these under sourcetree.md
-#- [pageserver/README.md](/pageserver/README.md)
+#- [pageserver-services.md](./pageserver-services.md)
+#- [walservice.md](./walservice.md)
```
Comment (hlinnaka) [inline docs/SUMMARY.md:56]: These can be just removed, they are in the Table of Contents above already
```suggestion
```

**Review by hlinnaka — COMMENTED** [review]
> Updated. Can you take look? I'm not really familiar with the renaming. I only made the changes based on https://github.com/neondatabase/neon/commit/0b14fdb0783b745fbb99e08d49da1253db2b0134.

Thanks, LGTM. I mean, there's clearly a lot more we could do to clean up the docs, but this is a small step in the right direction.

> I don't see any references nor renaming of ./src/remote_storage/README.md. Do u have any ideas?

Hmm, it was removed in commit 3e6087a12f26ebefe6b91ea78be5d927c72b2a48. I think we can just remove that sentence with the reference to it. Would be nice to have more docs for the remote-storage code, of course, but that's a separate story..

**Pull request merged**